src/ApiClient/util/AuthController.ts

```diff
@@ -21,19 +22,22 @@ export type LoginErrors =
     | ErrorCode.LOGIN_RATELIMIT
     | ErrorCode.LOGIN_LOGGING_IN;
 
-export default new (class AuthController {
+interface IEvents {
+    loggingInDoneEvent: (status: InternalStatus<null, LoginErrors>) => void;
```

Comment:
Feel free to ignore this as it doesn't matter in the grand scheme of things, but events dont need to be postfixed with event